docs/AUTH_PROTOCOL.md

```diff
@@ -0,0 +1,393 @@
+    - Arguments:
+        - `primary`: When true, this is a primary authentication mechanism
+        - `name`: Display name of the security token to insert. Optional. For example, if an enterprise deploys only Yubikeys, it might be helpful for the end-user if the prompt asks them to insert their "Yubikey" rather than a generic "Insert security token" message.
+    - Response: frontend emits the `passkeyInserted` event when a passkey is inserted
+
+- `passkey`: FIDO security token handling
+    - The frontend prompts the user for a "pin" (which is actually an alphanumeric password)
+    - _After_ the frontend sends the response, the UI should prompt the user to touch their FIDO token
+    - The frontend shall continue monitoring for FIDO token insertion/removal events
+    - This method is always primary
+    - Arguments:
+        - `remainingAttempts`: The number of remaining attempts for the FIDO PIN until the device is locked
+    - Response:
+        - if the user has confirmed their entered pin code, then the frontend prompts the user to touch the FIDO token and emits a `response` event with the following arguments:
+            - `pin`: The user's entered pin code
+        - if the FIDO token is removed before the user is done interacting with the UI, the frontend emits a `passkeyRemoved` event
+        - Once the user touches the FIDO token, the authenticator will continue the converstaion on its own
+
+- `monitorSmartcard`: Prompt for the user to insert a smartcard
+    - The frontend monitors for smartcard insertion events, and prompts for the user to insert a smartcard
+    - If a smarcard is already inserted, the frontend responds immediately
+    - This method is either primary or secondary
+    - A frontend shouldn't advertise support for this mechanism in `supportedMechanisms`; support is implied for any frontend that supports the `smartcard` mechanism.
+    - Arguments:
+        - `primary`: When true, this is a primary authentication mechanism
+    - Response: frontend emits the `smartcardInserted` event when a smartcard is inserted
+
+- `smartcard`: Smart card handling
+    - Smartcards might have more than one certificate available. If this card has more than one certificate, the frontend first prompts the user to pick a certificate
+    - Next, the frontend prompts the user for a "pin" (which is actually an alphanumeric password)
+    - The frontend shall continue monitoring for smartcard insertion/removal events
+    - This method is always primary
+    - Arguments:
+        - `certificates`: An array of objects that describe the available certificates that the user can select between. Each object in the array has the following keys:
+            - `id`: The machine-usable string that identifies this certificate
+            - `slotName`: A display name for the PIV slot of this certificate. Example: "Certificate for PIV Authentication"
```

Comment:
This is the PKCS#11 certificate label. Lets not tie this to the PIV specific naming (example is ok).